src/Wallet/Wallet.ts

```diff
@@ -0,0 +1,30 @@
+import { Bip44AccountPublic } from 'cardano-wallet'
+import { TransactionOutput } from '../Transaction'
+import { AddressType } from '.'
+import { verifyMessage } from './lib/verify_message'
+
+export function Wallet (account: Bip44AccountPublic) {
+  return {
+    getNextReceivingAddress: () => getNextReceivingAddress(account),
+    getNextChangeAddress: () => getNextChangeAddress(account),
+    balance: () => balance(account),
+    balanceTransaction: (outputs: TransactionOutput[]) => balanceTransaction(account, outputs),
+    verifyMessage: (verificationArgs: { addressType: AddressType, signingIndex: number, message: string, signatureAsHex: string }) => verifyMessage(account, verificationArgs)
+  }
+}
+
+function getNextReceivingAddress (_account: Bip44AccountPublic) {

```

Comment:
I prefer seeing these functions just defined as properties of `Wallet` return object